File: examples/module.py
#
# jython examples for jas.

from jas import PolyRing, QQ, Module
#, SubModule

# module example

p = PolyRing(QQ(),"u,v,l", PolyRing.lex);
#r = Module( "Rat(u,v,l) L", cols=4 );
r = Module( "", p, cols=4 );
print "Module: " + str(r);
print;

G = r.gens();
print "gens() = ", [str(e) for e in G];

L = [ e.elem.val for e in G ]
print "gens() = ", [str(e) for e in L];

M = r.submodul( list=L );
print "M = ", M;

P = M.mset.getPolynomialList();
print "P = ", P.toScript();

print "M.isGB(): ", M.isGB();

File: examples/module.rb
#
# jruby examples for jas.
# $Id$
#

require "examples/jas"

# module example

p = PolyRing.new(QQ(),"u,v,l", PolyRing.lex);
#r = CommutativeModule.new( "Rat(u,v,l) L", nil, 4 );
r = CommutativeModule.new( "", p, 4 );
puts "Module: " + str(r);
puts;

G = r.gens();
puts "gens() = " + str( G.map { |e| str(e) }.join(", ") );
puts

L = G.map { |e| e.elem.val }
puts "gens() = " + str( L.map { |e| str(e) }.join(", ") );
puts

M = r.submodul("", L );
puts "M = " + str(M);
puts

P = M.mset.getPolynomialList();
puts "P = " + str(P.toScript());
puts

puts "M.isGB(): " + str(M.isGB());
puts
